roles/ceph-osd/tasks/scenarios/collocated.yml

```diff
@@ -55,11 +55,9 @@
     - dmcrypt
     - ceph_release_num.{{ ceph_release }} < ceph_release_num.luminous
 
-# NOTE (alahouze): if the device is a partition, the parted command below has
-# failed, this is why we check if the device is a partition too.
-- name: automatic prepare "{{ osd_objectstore }}" osd disk(s) without partitions with collocated osd data and journal
-  command: "ceph-disk prepare {{ ceph_disk_cli_options }} /dev/{{ item.key }}"
-  register: prepared_osds
+- name: generate device list when osd_auto_discovery
+  set_fact:
+    devices: "/dev/{{ item.key }}"
```

Comment:
👎 This won't work, this is not creating a list but just the last item of the list.